### DISCUSSION TITLE
Incorporate tracking of QR codes and subsequent notifications

### DISCUSSION THREAD

**Discussion by LeCanardQuoi** [cds-snc/covid-alert-documentation#45] [Ideas] [answered]
# Summary | Résumé

As a user, I wish to better track not only my problematic proximity contacts (#1304) but also the public places I have visited where my risks are presumably increased. I wish to support businesses that have invested in QR code upgrades during the pandemic, help them lessen their burden in recording my visit and encourage other businesses to similarly invest. Also applicable to public buildings such as schools, public libraries, hotels, flights, etc. Finally I wish to be notified sooner rather than later via the app rather than the problematic contact tracing, should a propagation be linked to that business. 

# Design detail | Précisions au sujet de la conception

The app would record a QR encounter as an event much similar to a proximity contact and report event (including actual business) in a daily log. The event could be temporarily notified by the app to the screen. Should a propagation be linked to a businesss I have frequented over a specific period, my app should be notified by the app server similarly to diagnosed contacts reporting back.

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

The proposed enhancement is not expected to infringe on personal privacy (or individual freedom). Yes, the user of the app will be aware that visit is recorded similarly to an app “proximity contact”. Lesser even than signing guest register for people without the app. This proposed enhancement would make the app more useful, lessen the burden on existing contact tracing and help drive toward a higher adoption rate. This request does not cover the app server part. Business could be expected to run the app and report back a diagnosed case to the system. Yes this enhancement request also repurposes the app similarly to social distancing cds-snc/covid-alert-app#1304 and reporting localized app adoption rate/ 3 digits postal code cds-snc/covid-alert-app#1305. It could be reviewed as part of a relaunch driving towards higher adoption rates in response to a new virus strain propagating faster, hence requiring a stronger app solution.  And let's remember lessons from #Ebola when reviewing: "Moving quickly is essential" and avoid aiming for perfection before making a move”.

**Comment by timarney** [comment]
Ref: cds-snc/covid-alert-app#1384

**Comment by courtneycl** [comment, marked as answer]
Hi @LeCanardQuoi - thank you for your comments. Please refer to [this doc](https://github.com/cds-snc/covid-alert-documentation/blob/main/product/qr-codes.md) for an overview of our work on QR codes and notifications for venue outbreaks. We'll be updating documentation and details in that repo as we go forward.

**Comment by LeCanardQuoi** [comment]
> Hi @LeCanardQuoi - thank you for your comments. Please refer to [this doc](https://github.com/cds-snc/covid-alert-documentation/blob/main/product/qr-codes.md) for an overview of our work on QR codes and notifications for venue outbreaks. We'll be updating documentation and details in that repo as we go forward.

Am delighted. Will however require careful thought about a phased deployment (release plan), including field trial, public policy support and ... messaging. First step would be to simply record these QR scan (whether "partner" in program or not) as an other form of proximity events (rf. cds-snc/covid-alert-app#1304) and make core users aware of such logging. Much similar to logging proximity events imho.  Most useful info for stix&stones contact tracing people (rf. cds-snc/covid-alert-app#1395). Final notification remains similarly elusive imho but doable with sufficient pragmatism, proper kpi metrics (rf cds-snc/covid-alert-app#1305), community engagement and less wokeism ;-) Must learn to walk before running. Microsoft wisdom of "eating own dogfood" most applicable here. Deploy first in federal "facilities": custom&immig, airline flight, post office and Service Canada points of service. Great pressure for BC and Alta btw to get off their derrière. Hence public policy absolutely vital to deploy successfully, and "turbocharge" these adoption/retention rates. Cheers to the Marmot!

**Comment by LeCanardQuoi** [comment]
Still not sure on process to exchange on related requirements. Two additional sub-reqs: 1) Tracking as proximity events non signatory QR scan code sites would have merits for an app user wishing to track their whereabouts (without depending on Google map trace). Would be useful before more comprehensive support gets off the ground, likely involving bureaucratic delays. 2) Make sure the feature can be individually activated and deactivated, similarly to activating/deactivating Bluetooth proximity events, and log these commands to the Google setting log.

**Comment by obrien-j** [comment]
In keeping with our newly really release cds-snc/covid-alert-documentation#31 practices, I've moved this up into the covid-alert-documentation repo as our central point for taking input from folks outside the team, and am converting the issue into a Discussion.

**Reply by LeCanardQuoi** [reply]
No updates in one month in the discussion area. Is launching major new feature still planned? Would be great to know if it can be released before reopening up at the end of looming lockdown. Plan "B" stuff...
